docs: update client sdk get evaluation details (#155)
bucketeer-io/bucketeer-docs: docs/sdk/client-side/flutter/index.md

This is useful if you use another A/B Test solution with Bucketeer and need to know the variation name, reason, and other information.

:::caution Deprecated

The `evaluationDetails` is deprecated. Please use the following [interfaces](#interface).

:::

#### Interfaces

<Tabs>
  <TabItem value="dart" label="Dart">

```dart showLineNumbers
Future<BKTEvaluationDetails<bool>> boolVariationDetails(
  String featureId, {
  required bool defaultValue,
});

Future<BKTEvaluationDetails<String>> stringVariationDetails(
  String featureId, {
  required String defaultValue,
});

Future<BKTEvaluationDetails<int>> intVariationDetails(
  String featureId, {
  required int defaultValue,
});

Future<BKTEvaluationDetails<double>> doubleVariationDetails(
  String featureId, {
  required double defaultValue,
});

Future<BKTEvaluationDetails<BKTValue>> objectVariationDetails(
  String featureId, {
  required BKTValue defaultValue,
});
```

  </TabItem>
</Tabs>

#### Object

<Tabs>
  <TabItem value="dart" label="Dart">

```dart showLineNumbers

> class BKTEvaluationDetails<T> {
>   final String featureId;     // The ID of the feature flag.
>   final int featureVersion;   // The version of the feature flag.
>   final String userId;        // The ID of the user being evaluated.
>   final String variationId;   // The ID of the variation assigned.
>   final String variationName; // The name of the variation assigned.
>   final T variationValue;     // The value of the variation assigned.
>   final Reason reason;        // The reason for the evaluation.
> 
>   enum Reason {
>     TARGET,                   // Evaluated using an Individual targeting.
>     RULE,                     // Evaluated using a custom Rule targeting.
>     DEFAULT,                  // Evaluated using the Default Strategy.
>     CLIENT,                   // The flag is missing in the cache. The default value was returned.
>     OFF_VARIATION,            // Evaluated using the Off Variation.
>     PREREQUISITE,             // Evaluated using a Prerequisite targeting.
>   }
> }

Reviewer: ```suggestion
@immutable
class BKTEvaluationDetails<T extends Object> {
  final String featureId;     // The ID of the feature flag.
  final int featureVersion;   // The version of the feature flag.
  final String userId;        // The ID of the user being evaluated.
  final String variationId;   // The ID of the variation assigned.
  final String variationName; // The name of the variation assigned.
  final T variationValue;     // The value of the variation assigned.
  final String reason;        // The reason for the evaluation.
}
```

```
  </TabItem>
</Tabs>

#### Usage

<Tabs>
  <TabItem value="dart" label="Dart">

```dart showLineNumbers

> const client = BKTClient.instance;

Reviewer: ```suggestion
final client = BKTClient.instance;
```
